crates/composefs/src/splitstream.rs

```diff
@@ -19,8 +19,31 @@ use crate::{
     util::{read_exactish, Sha256Digest},
 };
 
+pub const SPLITSTREAM_MAGIC: [u8; 7] = [b'S', b'p', b'l', b't', b'S', b't', b'r'];
+
 #[derive(Debug, FromBytes, Immutable, IntoBytes, KnownLayout)]
 #[repr(C)]
+pub struct SplitstreamHeader {
+    pub magic: [u8; 7], // Contains SPLITSTREAM_MAGIC
+    pub algorithm: u8,
+    pub content_type: u64, // User can put whatever magic identifier they want there
+    pub total_size: u64,   // total size of inline chunks and external chunks
+    pub n_refs: u64,
+    pub n_mappings: u64,
+    // Followed by n_refs ObjectIDs, sorted
+    // Followed by n_mappings MappingEntry, sorted by body
+    // Followed by zstd compressed chunks
+}
+
+#[derive(Clone, Debug, FromBytes, Immutable, IntoBytes, KnownLayout)]
+#[repr(C)]
+pub struct MappingEntry {
+    pub body: Sha256Digest,
```

Comment:
If we're changing the format here...I think it'd be nice to make this one extensible.

However...bigger picture there's another consideration: There's obviously a metric *ton* of binary serialization formats out there. A custom one isn't wrong necessarily but...how about say [CBOR](https://www.rfc-editor.org/rfc/rfc8949.html) ? It has some usage and a proper RFC etc.

I guess a dividing line is "do we care about mmap()"? Probably not?

Reply:
splitstreams are essentially thin wrappers of existing binary formats (tars, ostree objects, etc), adding just references to other composefs repo objects. I'm not sure its overly helpful to use a complicated binary format for the wrapping, especially one which is completely different from the inner format.

That said, I agree that we should make it at least a bit extensible. This MR adds a magic header, but also adding a version field and a few  bytes of unused/unparsed space does seem quite useful.

Reply:
Although what about the algorithm agility? There's been some thoughts that for post-quantum crypto we may need to get away from sha256 in theory as far as I understand things.